packages/react-composites/src/mocks/useFakeChatAdapters.ts

```diff
@@ -170,11 +170,14 @@ const registerChatThreadClientMethodErrors = (
   chatThreadClientMethodErrors?: Partial<Record<keyof ChatThreadClient, _ChatThreadRestError>>
 ): void => {
   for (const k in chatThreadClientMethodErrors) {
-    chatThreadClient[k] = () => {
-      throw new RestError(chatThreadClientMethodErrors[k].message ?? '', {
-        code: chatThreadClientMethodErrors[k].code,
-        statusCode: chatThreadClientMethodErrors[k].statusCode
-      });
-    };
+    if (k in chatThreadClient) {
+      const key = k as keyof Omit<ChatThreadClient, 'threadId'>;
+      chatThreadClient[key] = () => {
```

Comment:
just wondering if we could use `k` here instead?

Reply:
We could duplicate the type cast everywhere but it seemed simpler to create a local automatic of the correct type and use that.